### PR TITLE
fix: only add '-Ctarget-feature=+sse4.1,+sse4.2' flags for x86_64

### PR DESCRIPTION
**Pull request by pugachAG**
With #8284 we upgraded cargo version to `1.66`, so https://github.com/rust-lang/cargo/pull/11114 is available. This PR is the same as #7130, but we can safely merge it now.

See #7650 for more context.

In order to test it I've used the code from [`librocksdb-sys/build.rs`](https://github.com/rust-rocksdb/rust-rocksdb/blob/master/librocksdb-sys/build.rs#L111) as part of `neard/build.rs`:
```
 fn main() {
-    if let Err(err) = try_main() {
-        eprintln!("{}", err);
-        std::process::exit(1);
+    let target_feature = std::env::var("CARGO_CFG_TARGET_FEATURE").unwrap();
+    let target_features: Vec<_> = target_feature.split(',').collect();
+    if target_features.contains(&"neon") {
+        panic!("FAIL: {target_feature}");
+    } else {
+        panic!("OK: {target_feature}");
     }
 }
```

It was was tested on M1 macbook, so I've changed the feature to `neon` (which is enabled by default), my `.cargo/config.toml` is as follows:
```
-[target.'cfg(target_arch = "x86_64")']
-rustflags = ["-Ctarget-feature=+sse4.1,+sse4.2", "-Cforce-unwind-tables=y"]
+[target.'cfg(target_arch = "aarch64")']
+rustflags = ["-Ctarget-feature=-neon", "-Cforce-unwind-tables=y"]
```

`neon` was successfully removed when using `1.66` toolchain:
```
OK: crc,...,vh
```

while it was still present with `1.65`:
```
FAIL: aes,..,neon,...,vh
```

This proves that rustflags are passed as env variable to `build.rs` in `1.66`, which was not the case in `1.65`

### CODE REVIEW

**Comment by akhi3030** [comment]
Thanks for this @pugachAG!  Can we somehow make sure that the appropriate flags are still being passed to rocksdb?  I.e. we are not getting a repeat of https://github.com/near/nearcore/pull/7648.  I believe @marcelo-gonzalez had originally noticed a performance degradation so we could repeat his performance test as well.

**Comment by pugachAG** [comment]
@akhi3030 good point, I've aded my test plan to the PR description, please take a look.

**Review by akhi3030 — APPROVED** [review]

File: .cargo/config.toml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 [build]
 # We compile with `panic=abort`, so we need `-Cforce-unwind-tables=y`
 # to get a useful backtrace on panic.
-rustflags = ["-Ctarget-feature=+sse4.1,+sse4.2", "-Cforce-unwind-tables=y"]
\ No newline at end of file
+rustflags = ["-Cforce-unwind-tables=y"]
+
+[target.'cfg(target_arch = "x86_64")']
+rustflags = ["-Ctarget-feature=+sse4.1,+sse4.2", "-Cforce-unwind-tables=y"]
```
Comment (nagisa) [inline .cargo/config.toml:7]: Does the `force-unwind-tables` need to be duplicated?

Comment (pugachAG) [reply .cargo/config.toml:7]: yes, I've explicitly verified the emitted `rustc` commands.

also found the confirming issue: https://github.com/rust-lang/cargo/issues/5376

**Comment by marcelo-gonzalez** [comment]
just tried running `view-state apply` like I mentioned before and it looks all good. Actually is a lot faster after this commit

**Pull request merged**